internal/transaction/paxos.go

```diff
@@ -0,0 +1,53 @@
+// MIT License. Copyright (c) 2020 CQFN
+// https://github.com/cqfn/degitx/blob/master/LICENSE
+
+package transaction
+
+import (
+	"context"
+)
+
+// Ballot number of proposals in Paxos algorithm
+type Ballot uint16
+
+// Proposal for Paxos protocol
+type Proposal struct {
+	// Ballot number of the proposal
+	Ballot
+	// Proposer ID, it should be unique number for each
+	// proposer participating in the transaction.
+	Proposer uint16
+}
+
+// PxAcceptor - Paxos acceptor API for proposer
+type PxAcceptor interface {
+	// Prepare takes Paxos 1A message and sends 1B message to the proposer
+	Prepare(ctx context.Context, p Proposal) error
+	// Accept takes Paxos 2A message and sends 2B message to proposer
+	Accept(ctx context.Context, p Proposal, v Vote) error
+}
+
+// PxProposer - Paxos proposer API for acceptor
+type PxProposer interface {
+	// Promise is sent by acceptor as 1B message if acceptor
+	// was prepared succesffully, it includes a proposal ballot
+	// number. Promise means that acceptor promises to proposer
+	// to reject all next messages if ballot number of such message
+	// are less than ballot number of promise.
+	Promise(ctx context.Context, b Ballot) error
+
+	// Reject 1B message from acceptor. The acceptor sends such message
+	// if it already accepted a value from proposer. The reject message
```

Comment:
Acceptor rejects only if it already accepted a **different** value from other proposer. If it has accepted the **same** value, it could promise, while it still forms the quorum.

Reply:
@stepanvalyavskiy on 1st stage the acceptor doesn't know the value proposed, since the proposer sends only the ballot number without a value, so it can count only ballot numbers having already accepted and the ballot number from 1A message.

It seems this method is not really clear, I'll try to merge these two methods (promise and reject) into single one to correspond for Paxos 1B message receiving

Reply:
Ok agree, then it's only could be renamed according to Paxos terms . e.g:  `PromiseForced`